### PR TITLE
Fix end_of_round

### PR DESCRIPTION
**Pull request by GinoGalotti**
This change was lost during the rebase. Now, when people click on "End of round" button, all active ability card decks should draw an extra card. 

This will fix issue https://github.com/johreh/gloomycompanion/issues/32

---
This change is [Reviewable](https://reviewable.io/reviews/johreh/gloomycompanion/33)

### CODE REVIEW

**Comment by mwinckler** [comment]
I think in order to take this feature we need to re-brand the "end of round" button as a "start new round" button because you're not supposed to draw monster ability cards until after players have declared their own actions.

Everybody hates the shuffle icon anyway, so here's an opportunity. :)

File: logic.js [hunk 1]
```diff
@@ -526,6 +538,10 @@ function end_round()
         modifier_deck.clean_advantage_deck();
         reshuffle_modifier_deck(modifier_deck);
     }
+
+    visible_ability_decks.forEach( function(deck) {
+      deck.new_round();
```
Comment (mwinckler) [inline logic.js]: Looks like inconsistent indentation here?

File: logic.js [hunk 2]
```diff
@@ -198,6 +198,18 @@ function load_ability_deck(deck_definition)
       }
     }
 
+    deck.new_round = function() {
+        if (this.discard.length > 0)
+        {
+            draw_ability_card(this);
+            // Maybe the last card was a reshuffle.
+            if (this.discard == 0)
```
Comment (mwinckler) [inline logic.js]: Should this be `this.discard.length`?

Comment (GinoGalotti) [inline logic.js]: Agree and changing it, although it works with this.discard (javascript casts 0 == [])

**Comment by GinoGalotti** [comment]
Yes, I think it's a good excuse to change the icon. I unsuccessfully tried to lure someone from the office to make a simple one. The idea should be that it's an "End of round" icon, instead of a "New round" one; mainly because right now the only way we have to check if a deck is active or not is checking its discard. I'm thinking of adding a use_element on the top left corner to disable (grey scale) it, so when people kill a type of monster they just freeze the deck and keep using the "new round" button, but I haven't spent that much time on it yet. I would also like to add a "right column" on every ability deck with a +/- selector for the level of the monster.

TL;DR If we wait for me to design an icon, this PR will take a while longer :p
